dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNLoadInterfaceIIDM.cpp

```diff
@@ -0,0 +1,160 @@
+  staticParameters_.clear();
+  double P = InjectorInterfaceIIDM::getP();
+  double P0 = getP0();
+  double Q = InjectorInterfaceIIDM::getQ();
+  double Q0 = getQ0();
+  double SN = SNREF;
+  SN = 1.5 * sqrt(P * P + Q * Q);
+  SN = (SN < SNREF ? SNREF : SN);
+
+  staticParameters_.insert(std::make_pair("p_pu", StaticParameter("p_pu", StaticParameter::DOUBLE).setValue(P / SNREF)));
+  staticParameters_.insert(std::make_pair("p0_pu", StaticParameter("p0_pu", StaticParameter::DOUBLE).setValue(P0 / SNREF)));
+  staticParameters_.insert(std::make_pair("q_pu", StaticParameter("q_pu", StaticParameter::DOUBLE).setValue(Q / SNREF)));
+  staticParameters_.insert(std::make_pair("q0_pu", StaticParameter("q0_pu", StaticParameter::DOUBLE).setValue(Q0 / SNREF)));
+  staticParameters_.insert(std::make_pair("sn_pu", StaticParameter("sn_pu", StaticParameter::DOUBLE).setValue(SN / SNREF)));
+  staticParameters_.insert(std::make_pair("p", StaticParameter("p", StaticParameter::DOUBLE).setValue(P)));
+  staticParameters_.insert(std::make_pair("p0", StaticParameter("p0", StaticParameter::DOUBLE).setValue(P0)));
+  staticParameters_.insert(std::make_pair("q", StaticParameter("q", StaticParameter::DOUBLE).setValue(Q)));
+  staticParameters_.insert(std::make_pair("q0", StaticParameter("q0", StaticParameter::DOUBLE).setValue(Q0)));
+  staticParameters_.insert(std::make_pair("sn", StaticParameter("sn", StaticParameter::DOUBLE).setValue(SN)));
+
+  if (InjectorInterfaceIIDM::getBusInterface()) {
+    v0_ = InjectorInterfaceIIDM::getBusInterface()->getV0();
+
+    double teta = InjectorInterfaceIIDM::getBusInterface()->getAngle0();
+    staticParameters_.insert(std::make_pair("v_pu", StaticParameter("v_pu", StaticParameter::DOUBLE).setValue(v0_ / vNom_)));
+    staticParameters_.insert(std::make_pair("angle_pu", StaticParameter("angle_pu", StaticParameter::DOUBLE).setValue(teta * M_PI / 180)));
+    staticParameters_.insert(std::make_pair("v", StaticParameter("v", StaticParameter::DOUBLE).setValue(v0_)));
+    staticParameters_.insert(std::make_pair("angle", StaticParameter("angle", StaticParameter::DOUBLE).setValue(teta)));
+  } else {
+    v0_ = 0;
+    vNom_ = 0;
+    staticParameters_.insert(std::make_pair("v_pu", StaticParameter("v_pu", StaticParameter::DOUBLE).setValue(0.)));
+    staticParameters_.insert(std::make_pair("angle_pu", StaticParameter("angle_pu", StaticParameter::DOUBLE).setValue(0.)));
+    staticParameters_.insert(std::make_pair("v", StaticParameter("v", StaticParameter::DOUBLE).setValue(0.)));
+    staticParameters_.insert(std::make_pair("angle", StaticParameter("angle", StaticParameter::DOUBLE).setValue(0.)));
+  }
+  // attention to sign convention
+}
+
+void
+LoadInterfaceIIDM::setBusInterface(const shared_ptr<BusInterface>& busInterface) {
+  InjectorInterfaceIIDM::setBusInterface(busInterface);
+}
+
+shared_ptr<BusInterface>
+LoadInterfaceIIDM::getBusInterface() const {
+  return InjectorInterfaceIIDM::getBusInterface();
```

Comment:
Useless I think (same for getInitialConnect, getP, getQ)

Reply:
Discussion is folowing

Reply:
In that case getBusInterface() is needed because "pure virtual". Without the prefix it loops on itself.